linkify-item-names.user.js

```diff
@@ -132,6 +137,16 @@ function linkify_items_in_syndicate_campaign_rewards() {
     }
 }
 
+function store_item_params(slug) {
+    var item_type = $('.data-list>.type>span');
+    if (item_type.length && (item_type.html().toLowerCase() == 'armor' || item_type.html().toLowerCase() == 'weapon')) {
+        var piercing = $('.data-list>.piercing-damage>span').html();
+        var impact = $('.data-list>.impact-damage>span').html();
+        var energy = $('.data-list>.energy-damage>span').html();
+        localStorage.setItem(ls_prefix + slug, ' (P:' + piercing + ', I:' + impact + ', E:' + energy + ')');
```

Comment:
...right. This is fine for opponents' equipment, and saves a lot of time vs. repeated `JSON.parse(...)` calls. :-)

OTOH, if we also start showing info for the player's equipment (inventory, storage room, etc.), we may want to see Accuracy, as well as Is-Long-Range vs. Hand-to-Hand (incl. neither being set). We could append that after some delimiter (";", "|", etc.), so the opponent-equipment-info display can truncate at that delimiter while the player-equipment-info display can show the added info.

(Anyway, no action for now; just something to consider in subsequent updates.)

Reply:
Thanks for the comment, pretty much what I was thinking about when writing this :)

I explicitly wanted to have something that would

- be simple and fast to implement
- perform well
- have little localStorage footprint/overhead

I think this storage format does satisfy the above at the cost of its inflexibility. If we want to implement other functionality based on the information stored in localStorage, I'd rather change the format to JSON instead of engineering something new and shiny (which usually turns out to be a wheel anyway).

Reply:
more detailed item stats may be better as a mouseover (though perhaps less accessible)

Reply:
Fair enough. :-) It'll definitely help that the JSON that'd eventually be parsed would be tiny (limited to one item), and that the pages it's happening on aren't time-critical. (E.g., if the info were being shown while in the combat UI, perf would be more relevant.)